ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ParquetHiveSerDe.java

```diff
@@ -234,4 +251,161 @@ StructTypeInfo prune() {
+    GroupType gType = parquetType.asGroupType();
+    Type innerField = gType.getType(0);
+    if (innerField.isPrimitive() || innerField.getOriginalType() != null) {
+      return createArrayType(convertParquetTypeToFieldType(innerField));
+    }
+
+    GroupType innerGroup = innerField.asGroupType();
+    if (innerGroup.getFieldCount() != 1) {
+      return createArrayType(convertGroupType(innerGroup));
+    }
+
+    return createArrayType(convertParquetTypeToFieldType(innerGroup.getType(0)));
+  }
+
+  private String createDecimalType(int precision, int scale) {
+    // example: decimal(10, 4)
+    return serdeConstants.DECIMAL_TYPE_NAME + "(" + precision + "," + scale + ")";
+  }
+
+  private String convertLogicalType(Type type) throws SerDeException {
+    LogicalTypeAnnotation lType = type.getLogicalTypeAnnotation();
+    if (lType instanceof IntLogicalTypeAnnotation) {
+      return convertParquetIntLogicalType(type);
+    } else if (lType instanceof StringLogicalTypeAnnotation) {
+      return serdeConstants.STRING_TYPE_NAME;
+    } else if (lType instanceof DecimalLogicalTypeAnnotation) {
+      DecimalLogicalTypeAnnotation dType = (DecimalLogicalTypeAnnotation) lType;
+      return createDecimalType(dType.getPrecision(), dType.getScale());
+    } else if (lType instanceof MapLogicalTypeAnnotation) {
+      return convertParquetMapLogicalTypeAnnotation(type);
+    } else if (lType instanceof ListLogicalTypeAnnotation) {
+      return convertParquetListLogicalTypeAnnotation(type);
+    } else if (lType instanceof DateLogicalTypeAnnotation) {
+      // assuming 32 bit int
+      return serdeConstants.DATE_TYPE_NAME;
+    }
+    throw new SerDeException("Unhandled logical type "  + lType);
+  }
+
+  private String convertParquetTypeToFieldType(Type type) throws SerDeException {
+    if (type.getLogicalTypeAnnotation() != null) {
+      return convertLogicalType(type);
+    } else if (type.isPrimitive()) {
+      return convertPrimitiveType(type.asPrimitiveType());
+    }
+    return convertGroupType(type.asGroupType());
+  }
+
+  private FieldSchema convertParquetTypeToFieldSchema(Type type) throws SerDeException {
+    String columnName = type.getName();
+    String typeName = convertParquetTypeToFieldType(type);
+    return new FieldSchema(columnName, typeName, "Inferred from Parquet file.");
+  }
+
+  public List<FieldSchema> readSchema(Configuration conf, String file) throws SerDeException {
+      ParquetMetadata footer;
+      try {
+        footer = ParquetFileReader.readFooter(conf, new Path(file), ParquetMetadataConverter.NO_FILTER);
+      } catch (Exception e) {
+        throw new SerDeException("Failed to read parquet footer:", e);
+      }
+
+      MessageType msg = footer.getFileMetaData().getSchema();
+      List<FieldSchema> schema = new ArrayList<>();
+      for (Type field: msg.getFields()) {
+        schema.add(convertParquetTypeToFieldSchema(field));
+      }
+      return schema;
```

Comment:
For debuggability, you may want to add a DEBUG level log message for the inferred schema.